ash/src/vk/prelude.rs

```diff
@@ -63,6 +63,15 @@ pub unsafe trait TaggedStructure {
     const STRUCTURE_TYPE: vk::StructureType;
 }
 
+/// Implemented for every structure that extends base structure `B`. Concretely that means struct
+/// `B` is listed in its array of [`structextends` in the Vulkan registry][1].
+///
+/// Similar to [`TaggedStructure`] , all `unsafe` implementers of this trait must guarantee that
+/// their structure is layout-compatible [`vk::BaseInStructure`] and [`vk::BaseOutStructure`].
+///
+/// [1]: https://registry.khronos.org/vulkan/specs/latest/styleguide.html#extensions-interactions
+pub unsafe trait Extends<B> {}
```

Comment:
Fwiw I've not found a use/need for it yet, but we might want to consider the base struct "unsized" / object-safe too?

```suggestion
pub unsafe trait Extends<B: ?Sized> {}
```

Reply:
I'd be inclined not to for simplicity, since it's difficult to imagine `Extends<dyn Trait>` or `Extends<[T]>` being useful. Note that `dyn Extends<T>` is already valid.

Reply:
Yeah, `dyn Extends<T>` is valid, as checked by the new tests (and requested by users), but `Extends<dyn T>` is not though we don't have any trait bound `T` that would be useful in this place.